proposals/lip-0057.md

```diff
@@ -1928,22 +1928,21 @@ def afterTransactionsExecute(b: Block) -> None:
 
     # Block b is an end-of-round block. Need to update snapshot and select validators for next round.
     # This must be done after the properties related to missed blocks are updated.
-    currentRound = getRoundNumberFromHeight(height)
-    nextRound = currentRound + 1
-
     updateSnapshotSubstore(height)
 
+    nextRound = getRoundNumberFromHeight(height+1)
+
     # No updates in validators and BFT weights are performed during bootstrap period.
     if nextRound <= genesisDataStore.initRounds:
         return
     
     # Validator selection for next round.
     # Calculate the active validators for next round based on the snapshot taken two rounds ago.
-    validatorList = getActiveValidators(snapshotStore(currentRound-2), nextRound)
+    validatorList = getActiveValidators(snapshotStore(nextRound), nextRound)
```

Comment:
minor thing, but we could even clean up the interface now, and just pass `nextRound` as a parameter and get the snapshot there directly

Reply:
Good catch. It is one of the improvements that we coundn't see in the previous stage. I think this way is cleaner, so I proceeded making this change.